languages/go/sample-basics.go

```diff
@@ -0,0 +1,84 @@
+
+package main
+
+import "fmt"
+
+func main() {
+  fmt.Println("+++ Sample Basics in Go +++")
+
+  println("Generating sample slice of ints...")
+  sample_numbers := []int{1, 2, 3, 4, 5, 10, 100}
+  println(stringify(sample_numbers))
+
+  println("Generating sample slice of strings...")
+  sample_strings := []string{"abc", "123", "xyz"}
+  println(stringify(sample_strings))
+
+  println("Summing sample ints...")
+  sample_numbers_sum := sum(sample_numbers...)
+  println(stringify(sample_numbers_sum))
+
+  println("Concatenating sample strings...")
+  sample_strings_concat := concat(sample_strings...)
+  println(stringify(sample_strings_concat))
+
+  println("Building and running sample tests...")
+  type Test struct {
+    name string
+    expected interface{}
+    actual interface{}
+  }
+  tests := []Test{
+    Test{name: "sample_numbers_sum", expected: 125, actual: sample_numbers_sum},
+    Test{name: "sample_string_concat", expected: "abc123xyz", actual: sample_strings_concat},
+  }
+  type TestResult struct {
+    success bool
+    scenario Test
+    message string
+  }
+  for _, test := range tests {
+    result := TestResult{scenario: test}
+    if (test.actual == test.expected) {
+      result.success = true
+      result.message = "PASSED!"
+    } else {
+      result.success = false
+      result.message = concat("FAILED!")
```

Comment:
Let's remove this unnecessary concatenation. I was originally going to join a detailed fail message here, but I removed it.